cumulusci/core/config/tests/test_config.py

```diff
@@ -47,64 +47,78 @@
 from cumulusci.utils.yaml.cumulusci_yml import GitHubSourceModel, LocalFolderSourceModel
 
 
+class FakeConfig(BaseConfig):
+    foo: dict
+
+
 class TestBaseConfig:
     def test_getattr_toplevel_key(self):
-        config = BaseConfig()
+        config = FakeConfig()
         config.config = {"foo": "bar"}
         assert config.foo == "bar"
 
-    def test_getattr_toplevel_key_missing(self):
+    @mock.patch("warnings.warn")
+    def test_getattr_toplevel_key_missing(self, warn):
         config = BaseConfig()
         config.config = {}
         assert config.foo is None
+        assert len(warn.mock_calls) == 1
 
     def test_getattr_child_key(self):
-        config = BaseConfig()
+        config = FakeConfig()
         config.config = {"foo": {"bar": "baz"}}
         assert config.foo__bar == "baz"
 
+    def test_strict_getattr(self):
```

Comment:
Why did you change the existing tests to use FakeConfig? It makes it much less clear that the goal here is to test BaseConfig.

Reply:
Because BaseConfig doesn't have a property called "foo".